### PR TITLE
usb: samples: mass: fix FLASH overlay for nrf52840_pca10056

### LINKED ISSUE #14459: usb: samples: mass: doesn't build with FLASH overlay
**Describe the bug**
USB MSC sample doesn't build with `overlay-flash-disk.conf` for nrf52840_pca10056.

**To Reproduce**
- `cd samples/subsys/usb/mass`
-  `west build --board nrf52840_pca10056 -- -DOVERLAY_CONFIG=overlay-flash-disk.conf`

**Expected behavior**
Sample builds.

**Impact**
Low priority for now. Should be fixed before MSC is considered ready.

**Screenshots or console output**
```
zephyr/misc/generated/configs.c: In function ‘_ConfigAbsSyms’:
zephyr/misc/generated/configs.c:275:154: error: invalid suffix "x" on integer constant
 GEN_ABSOLUTE_SYM(CONFIG_DISK_FLASH_START, 0x);
                                                                                                                                                          ^ 
zephyr/misc/generated/configs.c:276:172: error: expected expression before ‘)’ token
 GEN_ABSOLUTE_SYM(CONFIG_DISK_FLASH_MAX_RW_SIZE, );
                                                                                                                                                                            ^
zephyr/misc/generated/configs.c:277:184: error: invalid suffix "x" on integer constant
 GEN_ABSOLUTE_SYM(CONFIG_DISK_FLASH_ERASE_ALIGNMENT, 0x);
                                                                                                                                                                                        ^ 
zephyr/misc/generated/configs.c:278:169: error: invalid suffix "x" on integer constant
 GEN_ABSOLUTE_SYM(CONFIG_DISK_ERASE_BLOCK_SIZE, 0x);
                                                                                                                                                                         ^ 
zephyr/misc/generated/configs.c:279:154: error: invalid suffix "x" on integer constant
 GEN_ABSOLUTE_SYM(CONFIG_DISK_VOLUME_SIZE, 0x);
                                                                                                                                                          ^ 
[25/131] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/printk.c.obj
ninja: build stopped: subcommand failed.
```

Seems like following Kconfig symbols are defined empty:
`DISK_FLASH_DEV_NAME`, `DISK_FLASH_START`, `DISK_FLASH_MAX_RW_SIZE`, `DISK_FLASH_ERASE_ALIGNMENT`, `DISK_ERASE_BLOCK_SIZE`, `DISK_VOLUME_SIZE`

**Environment (please complete the following information):**
 - OS: Linux Mint 18.3 x64
 - Toolchain: Zephyr SDK 0.10.0
 - Zephyr 6368ed4edf5c5ffbccf9f904efe836ef5ca5d84a

### PR DESCRIPTION
**Pull request by masz-nordic**
Adds required Kconfig options for flash driver.
Adds FatFS filesystem for flash disk.
Fixes #14459.

### CODE REVIEW

**Review by galak — DISMISSED** [review]
This needs to update samples/subsys/usb/mass/sample.yaml to include nrf52840_pca10056.

I'd also ask you drop the changes to ext/fs/fat/ff.c from this PR.

**Comment by masz-nordic** [comment]
@galak can you revisit?

File: samples/subsys/usb/mass/prj_nrf52840_pca10056.conf [hunk 1]
```diff
@@ -3,10 +3,24 @@ CONFIG_GPIO=y
...
+CONFIG_DISK_LOG_LEVEL_ERR=y
+
 CONFIG_DISK_ACCESS_RAM=y
-CONFIG_MASS_STORAGE_DISK_NAME="RAM"
+
+CONFIG_MASS_STORAGE_DISK_NAME="NAND"
+CONFIG_DISK_ACCESS_FLASH=y
+CONFIG_DISK_FLASH_DEV_NAME="NRF_FLASH_DRV_NAME"
+CONFIG_DISK_FLASH_START=0xf8000
+CONFIG_DISK_FLASH_MAX_RW_SIZE=4
```
Comment (nvlsianpu) [inline samples/subsys/usb/mass/prj_nrf52840_pca10056.conf]: CONFIG_DISK_FLASH_MAX_RW_SIZE can be safely increased. We have no limit for this value - It is possible to wite/read whole flash at once. Try use one of 256/1014/4096

**Review by nvlsianpu — CHANGES_REQUESTED** [review]

File: samples/subsys/usb/mass/prj_nrf52840_pca10056.conf (hunk 1 above)
Comment (masz-nordic) [reply samples/subsys/usb/mass/prj_nrf52840_pca10056.conf]: Changed to 4096.

**Review by nvlsianpu — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
@galak can you revisit this one please?

File: samples/subsys/usb/mass/prj_nrf52840_pca10056.conf [hunk 2]
```diff
@@ -3,10 +3,24 @@ CONFIG_GPIO=y
...
 CONFIG_USB_MASS_STORAGE_LOG_LEVEL_ERR=y
+CONFIG_DISK_LOG_LEVEL_ERR=y
+
 CONFIG_DISK_ACCESS_RAM=y
-CONFIG_MASS_STORAGE_DISK_NAME="RAM"
+
+CONFIG_MASS_STORAGE_DISK_NAME="NAND"
+CONFIG_DISK_ACCESS_FLASH=y
+CONFIG_DISK_FLASH_DEV_NAME="NRF_FLASH_DRV_NAME"
+CONFIG_DISK_FLASH_START=0xf8000
```
Comment (jfischer-phytec-iot) [inline samples/subsys/usb/mass/prj_nrf52840_pca10056.conf]: Where does it point? Please use DT_FLASH_AREA_STORAGE_OFFSET and maybe DT_FLASH_AREA_STORAGE_SIZE.

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: samples/subsys/usb/mass/prj_nrf52840_pca10056.conf (hunk 2 above)
Comment (carlescufi) [reply samples/subsys/usb/mass/prj_nrf52840_pca10056.conf]: @masz-nordic in order to do that use the python function as seen here:
https://github.com/zephyrproject-rtos/zephyr/blob/master/Kconfig.zephyr#L93

Comment (masz-nordic) [reply samples/subsys/usb/mass/prj_nrf52840_pca10056.conf]: @carlescufi Are those functions callable from `.conf` files? They are used exclusively in `Kconfig` files and when trying to use one I get an error:
```
Error: Aborting due to non-whitelisted Kconfig warning 'warning: the value
'$(dt_hex_val,DT_FLASH_AREA_STORAGE_OFFSET)' is invalid for DISK_FLASH_START (defined at
subsys/disk/Kconfig:75), which has type hex -- assignment ignored'. Note: If this warning doesn't
point to an actual problem, you can add it to the whitelist at the top of
/home/marcin/ncs/zephyr/scripts/kconfig/kconfig.py.
```

Comment (carlescufi) [reply samples/subsys/usb/mass/prj_nrf52840_pca10056.conf]: ah, no sorry. Those are only valid for `Kconfig` files, so I don't see any way of doing it other than hardcoding the value unless you can get `DISK_FLASH_START` to be correctly configured without using the `.conf` file?

Comment (carlescufi) [reply samples/subsys/usb/mass/prj_nrf52840_pca10056.conf]: >Where does it point? Please use DT_FLASH_AREA_STORAGE_OFFSET and maybe DT_FLASH_AREA_STORAGE_SIZE.

You can't do that in `.conf` files at all, you can only use constants to assign.

Comment (jfischer-phytec-iot) [reply samples/subsys/usb/mass/prj_nrf52840_pca10056.conf]: Sorry, did not think about it, any way to make more generic?

**Review by jfischer-no — APPROVED** [review]

File: samples/subsys/usb/mass/prj_nrf52840_pca10056.conf (hunk 2 above)
Comment (carlescufi) [reply samples/subsys/usb/mass/prj_nrf52840_pca10056.conf]: @jfischer-phytec-iot The only way to make it more generic is to not set it in the `.conf` but rather have it set automatically by a `defaul` in a `Kconfig` file. Unless that is possible, I would leave as-is.

**Pull request merged**